README.md

```diff
@@ -49,10 +49,12 @@ The Make and CMake build systems create three libraries: libmbedcrypto, libmbedx
 
 You need the following tools to build the library with the provided makefiles:
 
-* GNU Make or a build tool that CMake supports.
-* A C99 toolchain (compiler, linker, archiver). We actively test with GCC 5.4, Clang 3.8, IAR8 and Visual Studio 2013. More recent versions should work. Slightly older versions may work.
-* Python 3 to generate the test code, and to generate sample programs in the development branch.
+* GNU Make 3.82 or a build tool that CMake supports.
+* A C99 toolchain (compiler, linker, archiver). We actively test with GCC 5.4, Clang 6, IAR8 and Visual Studio 2013. More recent versions should work. Slightly older versions may work.
```

Comment:
My thinking here is that limiting the set of supported compilers reduces our maintenance burden, and also enables us to update our CI in the future without surprising any users by dropping or reducing testing on 3.8. I wouldn't expect too many users to be using clang 3.8 for new projects. But we could leave this at 3.8 if you think this isn't helpful.

Reply:
I added this sentence originally. My thinking was to let users know that if they're using a different compiler, they're more likely run into issues. But maybe this was a mistake. We didn't remove IAR when stopped running it on the CI, for example — so it isn't up-to-date.

We do not, and should not, say that other compilers are not supported. We promise that our code works with a C99 compiler, even if it isn't on this list. We don't promise that our code compiles without warnings with any particular compiler: compiler warnings are too flighty for that. If our code doesn't work with an “exotic” compiler, and it isn't because of a compiler bug or non-compliance or because the platform doesn't meet our assumptions (unfortunately not documented explicitly — things like all-bits-zero being a null pointer), we will fix it.

Given that we haven't been keeping this up-to-date, I propose to remove it. I think it's useful for users if we make this information public, but we could keep it in the version-independent repository instead. So here we should just state

> A C99 toolchain (compiler, linker, archiver).